csrc/python_frontend/fusion_cache.cpp

```diff
@@ -733,10 +760,16 @@ void FusionCache::deserialize(std::string filename) {
   max_fusions_ = fusion_cache_buffer->max_fusions();
 
   // 2. Deserialize fusions: (Fusion) and structure: (TrieNode) fields
-  std::generate_n(
-      std::back_inserter(fusions_),
-      fusion_cache_buffer->terminal_nodes()->size(),
-      [] { return std::make_unique<FusionSchedules>(); });
+  int64_t num_fusions = 0;
```

Comment:
Due to https://github.com/NVIDIA/Fuser/pull/3969/files#diff-b85e103b2d89336b517af402199164520e435b64550b3517eaa9df4df2a49e0cR695, `terminal_nodes()->size()` may be <= the max fusion ID. So I recovered the number of fusions from fusion IDs.